#! /bin/sh /usr/share/dpatch/dpatch-run
## 06-non-ASCII-folder-canvas.dpatch by  <julien.lavergne@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Upstream bug : http://bugzilla.gnome.org/show_bug.cgi?id=573014
## DP: Ubuntu bug : https://bugs.launchpad.net/ubuntu/+source/conduit/+bug/306236

@DPATCH@
diff -urNad conduit-fix~/conduit/modules/FileModule/FileModule.py conduit-fix/conduit/modules/FileModule/FileModule.py
--- conduit-fix~/conduit/modules/FileModule/FileModule.py	2009-02-24 22:56:27.000000000 +0100
+++ conduit-fix/conduit/modules/FileModule/FileModule.py	2009-02-24 23:03:47.000000000 +0100
@@ -116,7 +116,7 @@
         return self.folder
         
     def get_name(self):
-        return Vfs.uri_get_filename(self.folder)
+        return Vfs.uri_unescape(Vfs.uri_get_filename(self.folder))
 
     def _monitor_folder(self):
         self._monitor.add(self.folder, self._monitor.MONITOR_DIRECTORY)
